feat: Yield integration for DeFi

aabid-majid opened the pull request:
New `Earn` tab that replaces the `Staking` tab.
For now, the Earn tab will only have one data source: yield.xyz API.
However it is planned to host earn opportunities/positions for more data source, such as SEEK staking in the near future.
Therefore a clear separation between `Earn***` and `Yieldxyz***` components has been enforced.

Right now we are only scratching the surface when using yield.xyz capabilities, because its results are automatically trimmed down based on which products we have enabled in our yield.xyz org, and we did not enabled many.
No substrate/solana products are enabled, though they might in the future.
No products that require manual claiming, or that have cooldowns, are enabled. though they might in the future.
=> solution may appear a bit overengineered because of that, it has actually been tested from a dev org that has more capabilities. 

Misc improvements:
- new `WizardModalDialog` and `PopupSizeModalContainer` components that should be usable for all our wizards
- new `createGlobalOpenClose` utility that generates a `useGlobalOpenClose` that supports typed args

aabid-majid committed:
feat: show earn button on hover over token

aabid-majid committed:
feat: yield integration in extension-core

1. yield integration
2. earn button + product list from /v1/yields

aabid-majid committed:
feat: account selector page + modal

aabid-majid committed:
feat: correct logo in the product list

1. pass on the entire response payload in product list.
2. show composite logo in the product list.
3. correct header titles, etc.

rafiq-austinai committed:
feat: add Earn page with assets and discover tabs to dashboard
- Add EarnPage component with tabbed interface

rafiq-austinai committed:
Merge branch 'feat-deposit-from-portfolio' of https://github.com/TalismanSociety/talisman into feat-deposit-from-portfolio

aabid-majid committed:
feat: deposit amount for staking ui

aabid-majid committed:
fix: deposit modal/page enhancements

aabid-majid committed:
feat: complete backend deposit workflow

aabid-majid committed:
feat: transfer in progress for deposit

aabid-majid committed:
fix: get feedecimal token from the yield response

aabid-majid committed:
feat: complete deposit workflow & refactoring

rafiq-austinai committed:
feat(earn page): add staking positions display

aabid-majid committed:
refactor: separate folder for "Earn"

aabid-majid committed:
feat(deposit): stabilise modal layout

1. rename Earn modal to Product Selection Modal;
2. hide earn button from watch-only accounts
3. stabilise product selection modal layout

aabid-majid committed:
feat(earn): improve yield API integration and fix circular dependencies

- Change yield API query parameter from 'token' to 'inputTokens' for better filtering
- Fix BigInt conversion errors for decimal entry limits in useDepositFunds
- Prevent /enter API calls when amount is zero or validations fail
- Resolve circular dependency between useDepositFunds and useYieldTransaction
- Add useDepositValidation hook for centralized validation logic
- Update TypeScript types and fix linting errors
- Improve error handling and validation flow
- Remove duplicate DepositConfirmForm component
- Add new components: DepositDetails, DepositProgressBar, ConfirmDepositModal

Breaking changes:
- API now uses 'inputTokens' parameter instead of 'token'
- Removed DepositConfirmForm.tsx (replaced with new modal structure)

Fixes:
- Maximum call stack size exceeded error from circular dependencies
- BigInt conversion errors for decimal amounts like "0.1"
- Unnecessary API calls when validations fail

aabid-majid committed:
feat: styling for deposit and confirm modals

aabid-majid committed:
feat: cross icon in deposit modal

1. matching fonts in confirmdeposit modal.
2. bug fixes.

aabid-majid committed:
feat: tvl & network agnostic refactoring

- Fix button overlap in DashboardAssetRow and PopupAssetsTable
- Add TVL display to ProductItem with N/A fallback
- Fix balance display in EarnAccountsList (Planck to human-readable)
- Implement EarnFeeTooltip with network-agnostic fee estimation
- Refactor deposit transaction hooks for multi-network support
- Standardize EarnPillButton structure to match BondPillButton
- Update DepositProgressBar circle sizing and text styling

aabid-majid committed:
feat(earn): implement sequential Yield.xyz transaction processing with proper gas handling

- Add sequential transaction processing in YieldSubmitButton for allTransactions array
- Implement proper nonce calculation using api.ethGetTransactionsCount with incremental nonce
- Add Yield.xyz API integration: submitHash and pollStatus calls after each transaction
- Fix nonce calculation to use currentNonce + transactionIndex instead of Yield.xyz nonce
- Add 10% gas buffer to prevent transaction failures: gas + (gas * 10% / 100%)
- Fix ConfirmDepositModal progress step background styling to match normal modal
- Add proper type casting for account.address in ethGetTransactionsCount API call
- Implement transaction confirmation polling before proceeding to next transaction
- Add comprehensive error handling for Yield.xyz API calls
- Fix TypeScript errors related to nonce calculation and gas estimation

aabid-majid committed:
refactor(earn): simplify YieldSubmitButton loading state and add transaction filtering

aabid-majid committed:
fix(deposit): loader icon on submitting

aabid-majid committed:
feat(deposit): implement deposit workflow for DOT

- Move nonce calculation inside Ethereum branch only
- Add proper SignerPayloadJSON type casting for Polkadot
- Sync lastTransaction logic between Ethereum and Polkadot flows
- Ensure consistent transaction polling and callback behavior

aabid-majid committed:
fix: show earn button for a curated list of tokens

- refactor: move interfaces to types file

aabid-majid committed:
chore(Deposit): merge into talisman dev

aabid-majid committed:
fix: hide transaction priority row from yield opportunities

- yield api - '/enter' give its own gas/fee estimates, that's why removing priority

aabid-majid committed:
feat: earn tab fullscreen implementation

- fix: balances fetching logic
- feat: search field
- feat: styling

aabid-majid committed:
feat(earn tab): use yield api to fetch balances

- feat(deposit): filter out yields opps. with inputtokens not equal to selected token
- feat(earn tab): show input/output token logos in yield positions
- feat(earn tab): create a consolidated api response from 'balances' and 'yields' apis
- feat(earn tab): create backend and store as per architecture for 'earn tab'

aabid-majid committed:
fix(earn tab): styling

aabid-majid committed:
fix(earn tab): styling

aabid-majid committed:
fix(deposit): routing for popup on earn btn click

aabid-majid committed:
feat(earn tab): consistent earn ui for popup mode

- feat: update staking tile description
- feat: remove staking from navbars

aabid-majid committed:
fix(earn tab): styling

aabid-majid committed:
feat(earn tab): skeleton ui for popup mode

aabid-majid committed:
feat(earn tab): search functionality

- fix: don't show dummy logo if output token not present

aabid-majid committed:
fix(earn tab): show amount in fiat

aabid-majid committed:
feat(earn tab): show validator name for dot

aabid-majid committed:
refactor: reuse common function

aabid-majid committed:
feat(earn tab): filter positions based on selected account

aabid-majid committed:
feat(earn tab): yield position details

- feat: yield claim, add to position, dropdown menu
- feat: styling ui

aabid-majid committed:
feat: migrate custom yield apis to yieldxyz sdk

aabid-majid committed:
feat: yield sdk migration

aabid-majid committed:
feat: migration to yield sdk

aabid-majid committed:
fix: product selection shows none when validator also present

aabid-majid committed:
feat(deposit): use talisman's fee est for eth

aabid-majid committed:
feat: deposit workflow for solana

aabid-majid committed:
fix: show token/fiat balances with proper units

- feat: fetch cached balances while new ones are fetched

aabid-majid committed:
fix: set limit to 100 and filter inactive validtors

aabid-majid committed:
Merge branch 'dev' into talisman-aai-dev

aabid-majid committed:
chore: remove unused icons

aabid-majid committed:
fix: add missing @yieldxyz/sdk dependency to extension-core

aabid-majid committed:
fix: wrong icons in navbar

- fix: caching and fetching data in earn tab.

aabid-majid committed:
feat: use talisman's fee estimation for sol

feat: create new deserializer for hex

aabid-majid committed:
feat: better grouping of balances api response

aabid-majid committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops commented on apps/extension/src/inject/solana/util.ts:
```diff
@@ -54,3 +54,49 @@ export const deserializeTransaction = (transaction: string): Transaction | Versi
   }
 }
+
+export const deserializeTransactionFromBase64 = (
```
Please dont add code in the `extension/src/inject` folder: this folder is for code that is injected in all websites by the extension, to allow dapps to communicate with the wallet. it needs to be as tiny as possible.

looks like this should go in `packages/solana` instead

0xKheops commented on apps/extension/src/ui/apps/dashboard/layout/DashboardLayout.tsx:
```diff
@@ -71,10 +71,11 @@ const RESPONSIVE_FLEX_SPACING = classNames(
 const NavButton: FC<{
   label: ReactNode
   icon: FC<{ className?: string }>
+  activeIcon?: FC<{ className?: string }>
```
looks unused

0xKheops commented on apps/extension/src/ui/apps/popup/pages/Earn/DepositConfirm.tsx:
```diff
@@ -0,0 +1,194 @@
+        networkId: token.networkId as `0x${string}`,
+        payload: transaction.tx as
+          | import("@solana/web3.js").Transaction
+          | import("@solana/web3.js").VersionedTransaction,
```
is there a reason why there are inline imports here?

0xKheops commented on apps/extension/src/ui/apps/popup/pages/Earn/DepositConfirm.tsx:
```diff
@@ -0,0 +1,194 @@
+  const [progress, setProgress] = useState<{ networkId: string; txId: string } | null>(null)
+
+  // Get parameters from URL
+  const account = searchParams.get("account") || ""
```
a bit worried about a modal playing with the page's url, especially with generic names such as `tokenId` and `account` which could be used by the portfolio page itself in the future (or even now, would need to check)
is it possible to keep the wizard state variables in a react context or an observable that would be local to the modal ? if not, please prefix all url params with `earn`, such as `earnTokenId` and `earnAccount`

I imagine this was inspired by the Send Funds wizard, but the Send Funds wizard lives in its own popup page (also using url params there allow to open the page with prefilled parameters)

0xKheops reviewed (changes requested):
Would it be possible to rename "yield" to "earn" ?
it's very likely that the earn tab will have more providers (yieldxyz being just the first one), so it would be preferable to have consistency in the naming of what s related to the earn feature.
at first glance main ones to rename are: 
- extension-core/domains/yield folder package
- extension/ui/api methods and associated backend handlers
- hooks names

0xKheops committed:
feat: earn remote config (#2207)

aabid-majid committed:
feat: discover tab ui

- feat: use remote configured allowed tokens/networks for earn
- feat: discover tab hooks etc

aabid-majid committed:
feat: optimise yield products pagination and sorting

- refactor: Update API to fetch 100 items per call instead of 20
- feat: Add server-side sorting by reward rate in extension-core
- feat: Implement client-side pagination (20 items at a time, max 100)
- feat: Apply pagination to Discover tab, ProductSelectionModal, and
  ProductList components
- refactor: Remove API-based pagination props in favor of client-side logic
- fix: ValidatorPicker scroll container bottom padding
- refactor: Ensure consistent UX across all yield product displays
- refactor: Reduce API calls while maintaining responsive UI
- Hide "Show more" button after 100 items displayed
- Maintain existing product filtering and error handling

aabid-majid committed:
feat: implement discover tab deposit workflow with product-first flow

- feat: add tokenMapping utility to map YieldDto input tokens to chaindata tokenIds
- feat: add tokenId prop to DiscoverTokenRow and pass to ProductItem components
- feat: implement modal state mngmt in EarnOnYourAssets and discvr opp.
- feat: wire up ValidatorPicker, EarnAccountPicker, DepositModal, and ConfirmDepositModal
- feat: add status.enter filtering to match EarnPillButton workflow consistency
- fix: resolve BigInt conversion errors in deposit validation and yield transaction hooks
- feat: implement standard Polkadot fee calculation using SendFunds pattern
- fix: resolve TypeScript errors in YieldSubmitButton for transaction type compatibility
- feat: support both popup navigation and dashboard modal workflows

aabid-majid committed:
feat: hide search field when Discover tab is selected in Earn section

- fix: earn assets not displaying
- refactor: remove version bump in store

aabid-majid committed:
fix: exclude incompatible network/address pairs

aabid-majid committed:
fix: add yield supported networks

aabid-majid committed:
fix: multi step transactions not working

aabid-majid committed:
fix: error on opening transaction priority row

aabid-majid committed:
fix: show transfer in progress modal while depositing

aabid-majid committed:
feat: implement proper data storage/fetching for earn tab

 - feat: use getQuery$ to subscribe to store.

aabid-majid committed:
feat: simultaneous balance api calls for different accounts

aabid-majid committed:
feat: network selector modal/page when token exists on multiple networks

aabid-majid committed:
chore: point to proxy

aabid-majid committed:
feat: use token address instead of symbol in yield

aabid-majid committed:
feat: fetch and store all yield products for "discover" tab

- fix: move network logo after token symbol
- fix: set max-width to protocol value

aabid-majid committed:
feat: deposit on "add to position"

aabid-majid committed:
fix: shorten account address in account picker

aabid-majid committed:
fix: height issue in validator address

aabid-majid committed:
Merge branch 'dev' into talisman-aai-dev

aabid-majid committed:
feat: claim workflow completed

- fix: genesis hash not added in dot transaction execution in earn

aabid-majid committed:
feat: implement withdraw workflow for earn products via yield.xyz

aabid-majid committed:
fix: make account button unclickable in deposit form

aabid-majid committed:
fix: make account picker consistent with current design across the product

aabid-majid committed:
fix: styling in deposit confirm page/modal

- feat: give border to protocol value
- fix: network picker row in popup mode

aabid-majid committed:
fix: max button in deposit form

aabid-majid committed:
fix: make withdraw amount input field readonly

- fix: pill button for protocol
- fix: remove swap and max buttons in withdraw form

aabid-majid committed:
fix: error message ui

aabid-majid committed:
fix: token logo size in confirm deposit modal

aabid-majid committed:
fix: add yield type pill in the detail page

aabid-majid committed:
fix: removed withdraw button and hover effect from yield position row

aabid-majid committed:
fix: show total amount in fiat on 'defi' row and make it collapsible'

aabid-majid committed:
fix: progress/success modal padding

aabid-majid committed:
fix: claim button font size/weight

aabid-majid committed:
fix: grouping of balances per account

- fix: calculate gas of nonskipped transactions

aabid-majid committed:
feat: caching strategy in EARN same as portfolio's

aabid-majid committed:
fix: load cached data while api is loading

aabid-majid committed:
fix: dont use gas from yield response

- fix: revert caching changes for earn tab

aabid-majid committed:
fix: max-width to amount field in withdraw modal

- fix: show error when token mapping not available in withdraw

aabid-majid committed:
fix: border on discover yields rows

aabid-majid committed:
fix: correct color and size of expand/collapse icon

aabid-majid committed:
fix: border radius on position rows

aabid-majid committed:
feat: persist state across pages in earn tab

aabid-majid committed:
fix: consistent network token icon sizes across the product

aabid-majid committed:
feat: implement search in discover tab

aabid-majid committed:
fix: show rewards in position details

aabid-majid committed:
fix: alignment in earn tab

aabid-majid committed:
fix: font size/weight in earn tab

aabid-majid committed:
fix: yield type border width and account name font

aabid-majid committed:
feat: show balance type in position details page

aabid-majid committed:
fix: claim/add to position button styling

aabid-majid committed:
feat: make acc picker in earn like sendfunds one

aabid-majid committed:
fix: remove pill background from amount edit tokenlogo

aabid-majid committed:
fix: styling in deposit/claim/withdraw forms

aabid-majid committed:
fix: secondary color to fiat in available balance

aabid-majid committed:
feat: new design for claim form

aabid-majid committed:
fix: padding in account row in claim form

aabid-majid committed:
fix: modal navigation & account picker bugs

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
fix: build

0xKheops committed:
wip: cleanup

0xKheops committed:
wip: try to get rid of mapping tables

0xKheops committed:
fix: yield positions observable

0xKheops committed:
wip: earn assets tab loading

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
fix: loading states

0xKheops committed:
fix: duplicate keys

0xKheops committed:
refactor: share earn tabs

0xKheops committed:
chore: cleanup earn page & tabs

0xKheops committed:
refactor: rename yield to yieldxyz

0xKheops committed:
chore: rename stuff

0xKheops committed:
fix: dont enhance data on backend

0xKheops committed:
feat: rework opportunities

0xKheops committed:
feat: store for opportunities

0xKheops committed:
feat: consollidate observables

0xKheops committed:
feat: providers

0xKheops committed:
fix: dont reference inject code

0xKheops committed:
fix: remove unused

0xKheops committed:
refactor: rename

0xKheops committed:
feat: common earn positions list

0xKheops committed:
fix: ui adjustments

0xKheops committed:
feat: eligible total

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
feat: optimize observables

0xKheops committed:
feat: do less queries

0xKheops committed:
refactor: positions should not embed products

0xKheops committed:
fix: yap url

0xKheops committed:
refactor: move yieldxyz folder

0xKheops committed:
refactor: earn handler

0xKheops committed:
wip: deposit wizard

0xKheops committed:
wip: confirm page

0xKheops committed:
feat: tx eth

0xKheops committed:
feat: txs flow including nonce management

0xKheops committed:
feat: wizard modal dialog

0xKheops committed:
feat: product info on amount form

0xKheops committed:
feat: user friendly duration display

0xKheops committed:
chore: remove fee estimate from amount form

0xKheops committed:
refactor: share hooks

0xKheops committed:
fix: periods

0xKheops committed:
feat: transaction steps

0xKheops committed:
fix: stepper

0xKheops committed:
feat: fine tune forms

0xKheops committed:
feat: fine tune signing ux

0xKheops committed:
chore: cleanup

0xKheops committed:
fix: fee rows disappearing

0xKheops committed:
fix: flickering when chaining txs

0xKheops committed:
fix: wizard context

0xKheops committed:
feat: shared hook to manage a yieldxyz action's transactions series

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
feat: export yield argument schema

0xKheops committed:
feat: validation

0xKheops committed:
feat: set max

0xKheops committed:
feat: display tx error

0xKheops committed:
fix: show account chain badge

0xKheops committed:
refactor: rename and isolate yieldxyz specifics

0xKheops committed:
refactor: tmp rename

0xKheops committed:
refactor: rename

0xKheops committed:
feat: do not display error while processing

0xKheops committed:
feat: leverage aggregated positions endpoint

0xKheops committed:
feat: use online api

0xKheops committed:
feat: basic search in discover tab

0xKheops committed:
wip: positions page

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
wip: positions screen

0xKheops committed:
wip: exit wizard

0xKheops committed:
wip: claim modal

0xKheops committed:
feat: generic amount edit

0xKheops committed:
feat: exit flow

0xKheops committed:
chore: rename file

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
wip: claim and testing

0xKheops committed:
fix: fine tune opportunities

0xKheops committed:
feat: pending action wizard

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
feat: one route per tab

0xKheops committed:
wip: popup pages

0xKheops committed:
wip: popup screens

0xKheops committed:
fix: adjust positions tab for popup

0xKheops committed:
fix: ajust positions tab

0xKheops committed:
fix: popup position screen

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
feat: claim

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: changeset

0xKheops committed:
chore: cleanup

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
wip: refresh position

0xKheops committed:
wip: refresh position

0xKheops committed:
wip: position refresh

0xKheops committed:
wip: earn button dashboard

0xKheops committed:
feat: product picker

0xKheops committed:
feat: popup earn button

0xKheops committed:
chore: cleanup

0xKheops committed:
feat: risk analysis and fix positions obs

0xKheops committed:
chore: cleanup

0xKheops committed:
feat: use yap proxy

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
fix: import

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
fix: create open close

0xKheops committed:
chore: cleanup

0xKheops committed:
fix: ignore non-evm products

0xKheops committed:
fix: import

0xKheops committed:
feat: block actions for non-owned accounts

0xKheops committed:
feat: block wizards for external accounts

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
fix: exitable balance

0xKheops committed:
chore: cleanup

0xKheops committed:
chore: cleanup

0xKheops committed:
fix: truncated label

0xKheops committed:
Merge branch 'dev' into talisman-aai-dev

0xKheops committed:
fix: contain risk analysis drawer inside the modal

0xKheops committed:
fix: submit button state while waiting for api to find the tx
